src/loopfilter.rs

```diff
@@ -384,3 +395,726 @@ extern "C" {
+            let L = if (*l.offset(0))[0] as c_int != 0 {
+                (*l.offset(0))[0] as c_int
+            } else {
+                (*l.offset(-(1 as c_int) as isize))[0] as c_int
+            };
+            if !(L == 0) {
+                let H = L >> 4;
+                let E = (*lut).e[L as usize] as c_int;
+                let I = (*lut).i[L as usize] as c_int;
+                let idx = (*vmask.offset(1) & y != 0) as c_int;
+                loop_filter(
+                    dst,
+                    E,
+                    I,
+                    H,
+                    BD::pxstride(stride as usize) as isize,
+                    1 as c_int as ptrdiff_t,
+                    4 + 2 * idx,
+                    bd,
+                );
+            }
+        }
+        y <<= 1;
+        dst = dst.offset(4 * BD::pxstride(stride as usize) as isize);
+        l = l.offset(b4_stride as isize);
+    }
+}
+
+// TODO(perl) Temporarily pub until mod is deduplicated
+pub(crate) unsafe extern "C" fn loop_filter_v_sb128uv_c_erased<BD: BitDepth>(
+    dst: *mut DynPixel,
+    stride: ptrdiff_t,
+    vmask: *const u32,
+    l: *const [u8; 4],
+    b4_stride: ptrdiff_t,
+    lut: *const Av1FilterLUT,
+    w: c_int,
+    bitdepth_max: c_int,
+) {
+    loop_filter_v_sb128uv_rust(
+        dst.cast(),
+        stride,
+        vmask,
+        l,
+        b4_stride,
+        lut,
+        w,
+        BD::from_c(bitdepth_max),
+    )
+}
+
+unsafe fn loop_filter_v_sb128uv_rust<BD: BitDepth>(
+    mut dst: *mut BD::Pixel,
+    stride: ptrdiff_t,
+    vmask: *const u32,
+    mut l: *const [u8; 4],
+    b4_stride: ptrdiff_t,
+    lut: *const Av1FilterLUT,
+    _w: c_int,
+    bd: BD,
+) {
+    let vm: c_uint = *vmask.offset(0) | *vmask.offset(1);
+    let mut x: c_uint = 1 as c_int as c_uint;
+    while vm & !x.wrapping_sub(1 as c_int as c_uint) != 0 {
+        if vm & x != 0 {
+            let L = if (*l.offset(0))[0] as c_int != 0 {
+                (*l.offset(0))[0] as c_int
+            } else {
+                (*l.offset(-b4_stride as isize))[0] as c_int
+            };
+            if !(L == 0) {
+                let H = L >> 4;
+                let E = (*lut).e[L as usize] as c_int;
+                let I = (*lut).i[L as usize] as c_int;
+                let idx = (*vmask.offset(1) & x != 0) as c_int;
+                loop_filter(
+                    dst,
+                    E,
+                    I,
+                    H,
+                    1 as c_int as ptrdiff_t,
+                    BD::pxstride(stride as usize) as isize,
+                    4 + 2 * idx,
+                    bd,
+                );
+            }
+        }
+        x <<= 1;
+        dst = dst.offset(4);
+        l = l.offset(1);
+    }
+}
+
+#[cfg(all(feature = "asm", any(target_arch = "x86", target_arch = "x86_64")))]
+#[inline(always)]
+unsafe fn loop_filter_dsp_init_x86<BD: BitDepth>(c: *mut Rav1dLoopFilterDSPContext) {
+    use crate::include::common::bitdepth::BPC;
```

Comment:
I'd put this and the one in the arm `fn` under the `#[cfg(feature = "asm")]` imports.

Reply:
I think I've addressed this suggestion but I'm not 100% sure.